frontend/tests/playwright/product/product_image.spec.ts

```diff
@@ -0,0 +1,41 @@
+import { test, expect, Page } from '@playwright/test';
+
+test.describe('Product image test', () => {
+   test('product with a single image ', async ({ page }) => {
+      await page.goto('/products/iflex-opening-tool');
+
+      const image = page
+         .getByRole('img', { name: 'iFlex Opening Tool' })
+         .first();
+      expect(image).toBeVisible;
+   });
+
+   test('product with multiple images ', async ({ page }) => {
+      await page.goto('/products/repair-business-toolkit');
+
+      const firstImageSrc = await getCurrentActiveImgSrc(page);
+      const firstImage = page.locator(`img[src="${firstImageSrc}"]`).first();
+      await expect(firstImage).toBeVisible();
+
+      // Click on the arrow for the next image
+      await page.getByTestId('swiper-next-image').first().click();
+
+      const secondImageSrc = await getCurrentActiveImgSrc(page);
+      const secondImage = page.locator(`img[src="${secondImageSrc}"]`).first();
+      await expect(secondImage).toBeVisible();
+
+      // Assert that currently active image has different src than the first one
+      expect(firstImageSrc).not.toEqual(secondImageSrc);
+   });
+
+   /*
+    * Returns the src of the currently displayed active image
+    * Applies only for product variants with multiple images
+    */
+   async function getCurrentActiveImgSrc(page: Page) {
+      const divElement = await page.$('.swiper-slide.swiper-slide-active');
+      const imgElement = await divElement!.$$('img');
```

Comment:
I _think_ this could be something like this. 

```
return await page.locator('.swiper-slide-active').getByRole('img').getAttribute('src');
```

Might have to index into it still. Although I think locators automatically call `first()` if another locator is attached.

Reply:
The locator solution worked nicely and I had to use `first()`. It doesn't seem like it automatically calls `first()`.

Updated in https://github.com/iFixit/react-commerce/pull/1218/commits/acbd5dea522fe74827e26b5626c071636a4869ac to use the suggested changes above and got rid of the helper function.